advanced-achievements-plugin/src/main/java/com/hm/achievement/command/executable/InspectCommand.java

```diff
@@ -0,0 +1,166 @@
+
+/**
+ * Class in charge of displaying recipients of an achievement (/aach inspect).
+ *
+ * @author Rsl1122
+ */
+@Singleton
+@CommandSpec(name = "inspect", permission = "inspect", minArgs = 1, maxArgs = 5)
+public class InspectCommand extends AbstractCommand {
+
+	private static final int CACHE_EXPIRATION_DELAY = 60000;
+	private static final int PER_PAGE = 16;
+
+	private final AdvancedAchievements advancedAchievements;
+	private final AbstractDatabaseManager databaseManager;
+
+	private Map<String, Long> lastCached;
+	private Map<String, SupplierCommandPagination> cachedPaginations;
+
+	private final Map<String, String> achievementsAndDisplayNames;
+	private Map<String, String> achievementDisplayNamesAndDatabaseNames;
+
+	@Inject
+	public InspectCommand(@Named("main") CommentedYamlConfiguration mainConfig,
+			@Named("lang") CommentedYamlConfiguration langConfig, StringBuilder pluginHeader,
+			AdvancedAchievements advancedAchievements,
+			AbstractDatabaseManager databaseManager, Map<String, String> achievementsAndDisplayNames) {
+		super(mainConfig, langConfig, pluginHeader);
+		this.advancedAchievements = advancedAchievements;
+		this.databaseManager = databaseManager;
+		this.achievementsAndDisplayNames = achievementsAndDisplayNames;
+
+		this.lastCached = new HashMap<>();
+		this.cachedPaginations = new HashMap<>();
+	}
+
+	private Map<String, String> reverseMap(Map<String, String> map) {
+		Map<String, String> reversed = new HashMap<>();
+		for (Map.Entry<String, String> entry : map.entrySet()) {
+			reversed.put(entry.getValue().toLowerCase(), entry.getKey());
+		}
+		return reversed;
+	}
+
+	private int getPage(String[] args) {
+		return args.length > 1 && NumberUtils.isDigits(args[args.length - 1]) ? Integer.parseInt(args[args.length - 1]) : 1;
+	}
+
+	@Override
+	void onExecute(CommandSender sender, String[] args) {
+		// Argument Parsing
+		String achievementDisplayName = parseAchievementName(args).toLowerCase();
+		String achievementName = getAchievementName(achievementDisplayName);
+		if (achievementName == null) {
+			sender.sendMessage(ChatColor.RED + LangHelper.getReplacedOnce(CmdLang.ACHIEVEMENT_NOT_RECOGNIZED,
+					"NAME", achievementDisplayName, langConfig));
+			return;
+		}
+		int page = getPage(args);
+
+		advancedAchievements.getServer().getScheduler().runTaskAsynchronously(advancedAchievements, () -> {
+			// Cleaning the cache & caching desired pagination
+			cleanUpCache();
+			checkAndCache(achievementName);
+
+			// Send pagination
+			SupplierCommandPagination pagination = cachedPaginations.get(achievementName);
+			pagination.sendPage(page, sender);
+		});
+	}
+
+	private String parseAchievementName(String[] args) {
+		StrBuilder achName = new StrBuilder();
+
+		boolean lastArgumentIsNumber = args.length > 1 && NumberUtils.isDigits(args[args.length - 1]);
+		boolean secondLastArgumentIsNumber = args.length > 2 && NumberUtils.isDigits(args[args.length - 2]);
```

Comment:
I'm confused here, which would the second last arg be a number?

Reply:
If an achievement has a name like connect_1, the second last arg can be a number when viewing the second page eg. `/aach inspect connect 1 2`

In order to keep the page argument separate this boolean is needed.

----

I'll revisit this as it doesn't cover `/aach inspect connect 100` - Displaying page 100 for connect_100 achievement (if such exists)

Reply:
I did not touch this yet.